lib/surveyor/application.rb

```diff
@@ -2,9 +2,30 @@
 
 module Surveyor
   class Application < Dry::System::Container
+    boot(:rom, namespace: true) do |app|
+      init do
+        require "rom"
+
+        configuration = ROM::Configuration.new(:sql, 'postgres://localhost/surveyor_experiment')
+
+        register(:configuration, configuration)
+      end
+
+      start do
+        configuration = app['rom.configuration']
+
+        configuration.auto_registration(app.root.join('lib/surveyor'), namespace: true)
+        container = ROM.container(configuration)
```

Comment:
What is the point of registering the container after this point? I don't see anywhere where it is used besides inside this class, so I think leaving it as just a local variable would be enough... but I think if _you_ of all people did it then you probably have a very good reason for doing it! Enlighten me?

Reply:
@radar it's because the rom container is useful to get at as a named _thing_ later on. It'd let you do what I suggested in chat, auto-injecting the container into repos:

`lib/surveyor/repository.rb`:

```ruby
require "rom/repository"
require "surveyor/import"

module Surveyor
  class Repository < ROM::Repository::Root
    include Surveyor::Import.args["rom.container"]
  end
end
```

`lib/surveyor/survey_repo.rb`:

```ruby
require "surveyor/repository"

module Surveyor
  class SurveyRepo < Repository[:surveys]
  end
end
```

This gives you repo classes that can be auto-registered by dry-system, since the rom container is registered in app container and can be thusly auto-injected.

That make sense?

Reply:
@timriley Yes that all makes sense to me. Thanks for taking the time to explain :)